test/jdk/sun/security/mscapi/AllTypes.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import jdk.test.lib.Asserts;
+import jdk.test.lib.SecurityTools;
+
+import java.io.IOException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+
+/*
+ * @test
+ * @bug 6782021
+ * @requires os.family == "windows"
+ * @library /test/lib
+ * @summary More keystore types
+ */
+public class AllTypes {
+
+    public static void main(String[] args) throws Exception {
+        var nm = test("windows-my");
+        var nr = test("windows-root");
+        var nmu = test("windows-my-currentuser");
+        var nru = test("windows-root-currentuser");
+        var nmm = adminTest("windows-my-localmachine");
+        var nrm = adminTest("windows-root-localmachine");
+        Asserts.assertEQ(nm, nmu);
+        Asserts.assertEQ(nr, nru);
+    }
+
+    private static boolean hasAccess(String type) throws Exception {
+        try {
+            KeyStore ks = KeyStore.getInstance(type);
+            ks.load(null, null);
+            return true;
+        } catch (IOException ioe) {
+            if (ioe.getMessage().trim().endsWith("java.security.KeyStoreException: Access is denied.")) {
```

Comment:
Could the message be different if the system language is not English?

Reply:
I checked the API and its not guaranteed to be English.  I'm investigating a better method for detecting admin rights